lib/absinthe/type/built_ins/introspection.ex

```diff
@@ -51,28 +52,38 @@ defmodule Absinthe.Type.BuiltIns.Introspection do
       end
 
     field :on_operation,
+      deprecate: "Check `locations` field for enum value OPERATION",
       type: :boolean,
       resolve: fn
         _, %{source: source} ->
+          # TODO: support :query, :mutation, :subscription in `on`
           {:ok, Enum.member?(source.on, Absinthe.Language.OperationDefinition)}
       end
 
     field :on_fragment,
+      deprecate: "Check `locations` field for enum value FRAGMENT_SPREAD",
       type: :boolean,
       resolve: fn
         _, %{source: source} ->
+          # TODO: support :fragment_spread in `on`
           {:ok, Enum.member?(source.on, Absinthe.Language.FragmentSpread)}
       end
 
     field :on_field,
       type: :boolean,
+      deprecate: "Check `locations` field for enum value FIELD",
       resolve: fn
         _, %{source: source} ->
+          # TODO: support :field in `on`
           {:ok, Enum.member?(source.on, Absinthe.Language.Field)}
       end
 
+    field :locations, list_of(:__directive_location)
+
   end
 
+  enum :__directive_location, values: Type.Directive.valid_location_values
```

Comment:
@benwilson512 I'd like to piggyback off of the values defined in `Type.Directive.valid_location_values`, but of course the value of the attribute is quoted. We can't unquote all attr values (because we need to support resolvers, I figure); do you have a quick fix for this?

```
== Compilation error on file lib/absinthe/type/built_ins/introspection.ex ==
** (FunctionClauseError) no function clause matching in Absinthe.Type.Enum.Value.build/1
    lib/absinthe/type/enum/value.ex:31: Absinthe.Type.Enum.Value.build({{:., [line: 85], [{:__aliases__, [line: 85], [:Type, :Directive]}, :valid_location_values]}, [line: 85], []})
```

The tests won't pass after this is fixed -- there other issues -- but should get past this error.